src/app/shared/header/header.component.ts

```diff
@@ -28,43 +27,37 @@ export class HeaderComponent implements OnInit, OnChanges {
              private router: Router, private userDataService: UserDataService){}
 
   ngOnInit() {
-    const userData = this.userDataService.getUserData();
-    this.authService.subscribeOnLogin(() => {
-      this.isSignOutBtnVisible = true;
-    });
-    console.log('isSignOutBtnVisible', this.isSignOutBtnVisible); //false
-    if (userData) {
-      //console.log('userData VALUE', userData.token)
-      //console.log('isSignOutBtnVisible', this.isSignOutBtnVisible); //false
-
-      // NEW если токен есть то  через селект получить юзера
-      // this.store.pipe(
-      // select(selectUser))
-      // .subscribe(user => {
-      //   this.user = user;
-      //   console.log('USER', this.user);
-      // });
+    // 1 получить токен из стора
+   const tokenValueSubscription = this.store.pipe(
+      select(selectToken))
+      .subscribe(token => {
+        this.token = token;
+        console.log('TOKEN', this.token);
 
-      // ----- OLD
-      this.authService.getUserInfo(userData).subscribe(person => { // верни
-        this.user = person;
-        this.store.dispatch(getCurrentUser({user: person})); // сетим юзера
-        //console.log('USER', this.user); //USER { id, token..}
       });
 
-      // console.log('USER', this.user); // undef
-    }
+   const userValueSubscription = this.store.pipe(
+    select(selectUser))
+    .subscribe(user => {
+      this.user = user;
+      console.log('USER HEADER', this.user);
+    });
+
+   this.authService.subscribeOnLogin(() => {
+      this.isSignOutBtnVisible = true;
+    });
+   tokenValueSubscription.add(userValueSubscription);

```

Comment:
You add one more subscription, now `tokenValueSubscription` contains two subscriptions (user and token). Given that approach, `tokenValueSubscription` should be renamed to indicate that it contain multiple subscriptions. 
You can rename it to just `subscriptions`